html/sites/all/modules/hr/hr_api/hr_api.module

```diff
@@ -70,24 +55,17 @@ function hr_api_entity_hook($event, $entity, $type) {
     $wrapper = entity_metadata_wrapper($type, $id);
     $bundle = $wrapper->getBundle();
     foreach ($conf['hr_api_webhooks'] as $webhook) {
-      if (hr_api_should_call_webhook($webhook, $type, $bundle)) {
-        $call_webhook = TRUE;
-      }
-    }
-    if ($call_webhook) {
-      $langcode = $language->language;
-      $resource = hr_api_bundle_to_restful_plugin($bundle, $type);
-      if ($resource) {
-        $resource->setAccount($user);
-        $thandler = entity_translation_get_handler($type, $entity);
-        $translations = $thandler->getTranslations();
-        $translations = array_keys($translations->data);
-        $json_entities = json_decode(restful()->getFormatterManager()->format($resource->doGet($id), 'json'));
-        $json_entity = reset($json_entities->data);
-        foreach ($conf['hr_api_webhooks'] as $webhook) {
-          if (hr_api_should_call_webhook($webhook, $type, $bundle)) {
-            hr_api_call_webhook($webhook, $event, $resource, $langcode, $translations, $json_entity);
-          }
+      if ($restful_plugin = hr_api_should_call_webhook($webhook, $type, $bundle)) {
+        $langcode = $language->language;
+        $resource = restful()->getResourceManager()->getPlugin($restful_plugin);
+        if ($resource) {
+          $resource->setAccount($user);
```

Comment:
Is there a reason why all this has to be run as uid 1?

Reply:
I don't remember the exact reason, but I believe that when I implemented this without running as uid 1, it was not working depending on the permissions of the users (which is why I made it run as uid 1, to make sure I have the full json payload to send it to HID)